include/deal.II/base/function_signed_distance.h

```diff
@@ -254,6 +260,63 @@ namespace Functions
     private:
       const BoundingBox<dim> bounding_box;
     };
+
+
+    /**
+     * Signed-distance level set function of a Zalesak's disk proposed in
+     * @cite zalesak1979fully.
+     *
+     * It is calculated by the set difference $\psi(x) = \max(\psi_{S}(x),
+     * -\psi_{N}(x))$ of the level set functions of a sphere $\psi_{S}$ and a
+     * rectangle $\psi_{N}$. This function is zero on the surface of the disk,
+     * negative "inside" and positive in the rest of $\mathbb{R}^{dim}$.
+     *
+     * Contour surfaces of the signed distance function of a 3D Zalesak's disk
+     * are illustrated below:
+     *
+     * @image html signed_distance_zalesak_disk.png
+     *
+     * @ingroup functions
+     */
+    template <int dim>
+    class ZalesakDisk : public Function<dim>
+    {
+    public:
+      /**
+       * Constructor, takes the radius and center of the disk together
+       * with the width and the height of the notch.
+       *
+       * @param radius Radius of the disk.
+       * @param center Center of the disk.
+       * @param notch_width Width of the notch of the disk.
+       * @param notch_height Height of the notch of the disk.
+       */
+      ZalesakDisk(const Point<dim> &center,
+                  const double      radius,
+                  const double      notch_width,
+                  const double      notch_height);
+
+      /**
+       * Calculates the signed distance from a given point @p p to the disk.
+       * The signed distance is negative for points inside the disk, zero
+       * for points on the disk and positive for points outside the
+       * disk.
+       */
+      double
+      value(const Point<dim> & p,
+            const unsigned int component = 0) const override;
+
+    private:
+      /**
+       * Disk described by a sphere.
+       */
+      const Functions::SignedDistance::Sphere<dim> sphere;
+
+      /**
+       * Nothch described by a rectangle.
+       */
+      const Functions::SignedDistance::Rectangle<dim> notch;
```

Comment:
I have pushed a few minor simplifications. In particular, I have converted the notch to a `Function`. I think this is useful once we want to extract the level-set merging operations to helper functions.

@mschreter Could you rebase? I think the PR should be ready now! @simonsticko Could you take a second look?